week_1/dictionary_n_loops/driver.py

```diff
@@ -0,0 +1,92 @@
+import string
+import random
+import time
+
+import matplotlib.pyplot as plt
+
+from appending_random_string import (
+    append_random_string_items,
+    append_random_string_keys,
+    append_random_string_comprehension
+)
+
+
+def generate_random_dictionary(size: int) -> dict:
+    """
+    Helper function that generates a random dictionary
+    with random keys and random values
+    :param size: size of the required dictionary
+    :return: the dictionary object
+    """
+    rand_dict = dict()
+
+    for i in range(size):
+        key = ''.join(random.choice(string.digits + string.ascii_uppercase) for x in range(5))
+        value = ''.join(random.choice(string.digits + string.ascii_uppercase) for x in range(5))
+        rand_dict[key] = value
+
+    return rand_dict
+
+
+if __name__ == "__main__":
+
+    def calculate_execution_time(function):
+        """
+        Helper function that calculates the
+        execution time of a function.
+        :param function: function that you want to run.
+        :return: None
+        """
+        start = time.time()
+        function(random_dict, random_string)
+        end = time.time()
+        execution_time = end - start
+        return execution_time
```

Comment:
move this to utils.py as well. This is a utility function and would reside in a utils file.